tests/unit/definitions-with-nested-comments.test.scala

```diff
@@ -0,0 +1,73 @@
+// SYNTAX TEST "source.scala"
+
+   def // comment
+// ^^^ keyword.declaration.scala
+//     ^^^^^^^^^^ comment.line.double-slash.scala
+
+   def /* comment */
```

Comment:
What do you think about supporting things like `def /* test */ myDef()` at some point? I obviously don't want it for this PR, this is very low priority, but I'm just wondering. I think  it would require going to some specific state after keywords that are followed by an identifier.

Reply:
Indeed, low priority. There might be some useful code for the `end` pattern that might be useful for this.

Reply:
I'm not really sure if it is worth adding more complexity into the rule to support this syntax that nobody should be using anyway.